compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -779,6 +779,21 @@ object Contexts {
         }, gadts)
       }
 
+      // avoid recording skolems in lower bounds
+      // recording two skolem bounds results in an union, which is then simplified
+      // T >: Sko(U) | Sko(U) is simplified to T >: U, which is simply wrong
```

Comment:
Where is this simplification of `T >: Sko(U) | Sko(U)` to `T >: U` done? If this done only for Skolemtypes? I suspect the answer is that it happens for all singleton types. But then the fix should also be generalized to all singleton types, right?